<!-- -*- mode: XML -*- -->
<!-- run multi_hypothesis_tracking/multi_hypothesis_tracking_node in a node -->

<launch>

    <arg name="input_topic" default="/object_detections"/>

    <arg name="distance_threshold_for_hypotheses_merge" default="0.3"/>
    <arg name="max_covariance" default="5.0"/>
    <arg name="max_correspondence_distance" default="6.6"/>
    <arg name="kalman_covariance_per_second" default="0.5"/>
    <arg name="world_frame_id" default="odom"/>

    <arg name="hypothesis_age_threshold_in_seconds" default="0.0"/>
    <arg name="time_offset_for_predictions" default="0.05"/>

    <arg name="compute_likelihood" default="false"/>
    <arg name="subscribe_to_poses_only" default="false"/>

    <arg name="measure_time" default="false"/>

    <node pkg="multi_hypothesis_tracking" type="multi_hypothesis_tracking_node" name="multi_hypothesis_tracking_node"
          output="screen">

        <param name="input_topic" value="$(arg input_topic)"/>

        <param name="distance_threshold_for_hypotheses_merge" value="$(arg distance_threshold_for_hypotheses_merge)"/>
        <param name="max_covariance" value="$(arg max_covariance)"/>
        <param name="max_correspondence_distance" value="$(arg max_correspondence_distance)"/>
        <param name="kalman_covariance_per_second" value="$(arg kalman_covariance_per_second)"/>
        <param name="world_frame_id" value="$(arg world_frame_id)"/>

        <param name="hypothesis_age_threshold_in_seconds" value="$(arg hypothesis_age_threshold_in_seconds)"/>
        <param name="time_offset_for_predictions" value="$(arg time_offset_for_predictions)"/>

        <param name="compute_likelihood" value="$(arg compute_likelihood)"/>
        <param name="subscribe_to_poses_only" value="$(arg subscribe_to_poses_only)"/>

        <param name="measure_time" value="$(arg measure_time)"/>
    </node>

</launch>
